mlbstatsapi/mlbapi.py

```diff
@@ -20,53 +20,170 @@ def __init__(self, hostname: str = 'statsapi.mlb.com', ver: str = 'v1', logger:
-        # TODO Doc string
+        """
+        return the Team
+
+        Parameters
+        ----------
+        teamId : int
+            Team id 
+
+        Returns
+        -------
+        Team
+        """       
         mlbdata = self._mlb_adapter_v1.get(endpoint=f"teams/{teamId}")
+        
         if mlbdata and mlbdata.data['teams']:
             for team in mlbdata.data['teams']:
                 return Team(**team) 
 
     def get_team_id(self, teamName) -> List[int]:
-        # TODO Doc string
-        # teamName: Mariners
+        """
+        return a team Id
+
+        Parameters
+        ----------
+        teamName : str
+            Teams name
+
+        Returns
+        -------
+        List[ints]
+        """       
         mlbdata = self._mlb_adapter_v1.get(endpoint="teams") # Get all Teams
         teamIds = [] # create empty list
+        
         for team in mlbdata.data['teams']:
             if team['teamName'] == teamName: # find matching Team Name
                 teamIds.append(team['id']) # append match id
+        
         return teamIds
 
+    def get_team_roster(self, teamId) -> List[Player]:
+        """
+        return the team player roster
+
+        Parameters
+        ----------
+        teamId : int
+            Team id 
+
+        Returns
+        -------
+        List[Player]
+        """
+        mlbdata = self._mlb_adapter_v1.get(endpoint=f"teams/{teamId}/roster")
+        players = []
+        # if mlbdata is not empty, and 'roster' key is in mlbdata.data and mlbdata.data['roster] is not empty list
+        if mlbdata and ('roster' in mlbdata.data and mlbdata.data['roster']):
+            # for player in mlbdata.data['roster']
+            for player in mlbdata.data['roster']:
+                # remove person key and return value to person
+                person = player.pop('person')
+                # create Player, and unpack player and person
+                players.append(Player(**{**player, **person}))
+
+            return players
+
+    def get_team_coaches(self, teamId) -> List[Coach]:
+        """
+        return the team coach roster
+
+        Parameters
+        ----------
+        teamId : int
+            Team id 
+
+        Returns
+        -------
+        List[Coach]
+        """       
+        mlbdata = self._mlb_adapter_v1.get(endpoint=f"teams/{teamId}/coaches")
+        coaches = []
+
+        # if mlbdata is not empty, and 'roster' key is in mlbdata.data and mlbdata.data['roster] is not empty list
+        if mlbdata and ('roster' in mlbdata.data and mlbdata.data['roster']):
+            # for player in mlbdata.data['roster']
+            for coach in mlbdata.data['roster']:
+                # remove person key and return value to person
+                person = coach.pop('person')
+                # create Player, and unpack player and person
+                coaches.append(Coach(**{**coach, **person}))
+
+            return coaches
+
```

Comment:
Can you just walk me through `players.append(Player(**{**player, **person}))` and `coaches.append(Coach(**{**coach, **person}))`?

Reply:
It's a python3.5 syntax. It is unpacking and then merges the person and coach dictionaries, and then it unpacks and sends the results to the Player Class.

I had to do this for the following reason. A Player and Coach don't have a Person, they are a Person. 

The Player and Coach JSON looks like this -> 
```

{
"person": {
"id": 640462,
"fullName": "A.J. Puk",
"link": "/api/v1/people/640462"
},
"jerseyNumber": "33",
"position": {
"code": "1",
"name": "Pitcher",
"type": "Pitcher",
"abbreviation": "P"
},
"status": {
"code": "A",
"description": "Active"
},
"parentTeamId": 133
},
```

To pass the Person attributes, I need to "flatten" that dictionary so the id, full name, and link can be passed correctly to the Parent class.

Reply:
Ok gotcha that makes sense, since the issue was with Player and Coaches being Person. For a normal inheritance instance we would still be passing normal something like Game(**game) or venue(**venue)

Reply:
If the way this JSON is structured, probably not, but the only place I see inheritance being proper is in your game classes, but we'll see :)